main.py

```diff
@@ -0,0 +1,60 @@
+import argparse
+import asyncio
+import json
+from typing import Dict
+
+from gpt_promps import Gpt_promps
+from presentation_parser import PresentationParser
+from gpt_slide_explainer import GPTSlideExpander
+from slide_expander import SlideExpander
+
+
+def parse_args() -> argparse.Namespace:
+    """Parse command-line arguments.
+    Returns:
+        argparse.Namespace: Parsed command-line arguments.
+    """
+    argument_parser = argparse.ArgumentParser(description="Generate slide explanations from a PPTX file")
+    argument_parser.add_argument("pptx_file", help="Path to the PPTX file")
+    argument_parser.add_argument("-t", "--topic", help="Main topic of the presentation")
+    return argument_parser.parse_args()
+
+
+def output_to_json(output_path: str, expanded_slide_explanations: Dict[int, str]) -> None:
+    """Output the expanded slide explanations to a JSON file.
+
+    Args:
+        output_path (str): Path to the output JSON file.
+        expanded_slide_explanations (Dict[int, str]): Dictionary with slide numbers as keys and explanations as values.
+    """
+    json_to_save = {
+        "number of slides": len(expanded_slide_explanations.values()),
+        "explained slides": expanded_slide_explanations}
+    with open(output_path, "w") as file:
+        json.dump(json_to_save, file)
+
+
+def main():
+    args = parse_args()
+    presentation_parser = PresentationParser(args.pptx_file)
+    presentation_parser.parse()
+    slide_expander = GPTSlideExpander()
+
+    presentation_main_topic = args.topic
+
+    if not presentation_main_topic:
+        presentation_main_topic = slide_expander.generate_topic_for_presentation(
+            presentation_parser.get_all_presentation_text())
+
```

Comment:
You actuall logic shoudlnt be in your main method, the main method should handle input, and then call the actual logic. You want you IO and your logic separated as much as possible
https://wiki.c2.com/?SeparateIoFromCalculation